bundle/src/main/java/com/adobe/acs/commons/mcp/impl/ControlledProcessManagerImpl.java

```diff
@@ -64,6 +73,9 @@
     @Reference(cardinality = ReferenceCardinality.MANDATORY_MULTIPLE, bind = "bindDefinitionFactory", unbind = "unbindDefinitionFactory", referenceInterface = ProcessDefinitionFactory.class, policy = ReferencePolicy.DYNAMIC)
     private final List<ProcessDefinitionFactory> processDefinitionFactories = new CopyOnWriteArrayList<>();
 
+    @Reference(cardinality = ReferenceCardinality.OPTIONAL_MULTIPLE, bind = "bindScriptResolverService", unbind = "unbindScriptResolverService", referenceInterface = DynamicScriptResolverService.class, policy = ReferencePolicy.DYNAMIC)
```

Comment:
if the bind and unbind methods are as simple as here, I don't think that they are necessary. IIRC these implementations a kind of default (though I don't have the specific chapter of the OSGI spec handy).

Reply:
Yeah, I think so to but at some point during development I ran into a weird bug where optional services weren't binding and I needed these methods to debug it.  I think it's a hold-over but ultimately benign.